src/app/clusters/thermostat-server/thermostat-delegate.h

```diff
@@ -133,6 +134,30 @@ class Delegate
      *
      */
     virtual void ClearPendingPresetList() = 0;
+
+    virtual CHIP_ERROR GetScheduleTypeAtIndex(size_t index, Structs::ScheduleTypeStruct::Type & ScheduleType) = 0;
+
+    virtual uint8_t GetNumberOfSchedules() = 0;
+
+    virtual uint8_t GetNumberOfScheduleTransitions() = 0;
+
+    virtual DataModel::Nullable<uint8_t> GetNumberOfScheduleTransitionPerDay() = 0;
+
+    virtual CHIP_ERROR GetScheduleAtIndex(size_t index, ScheduleStructWithOwnedMembers & schedule) = 0;
+
+    virtual CHIP_ERROR GetActiveScheduleHandle(DataModel::Nullable<MutableByteSpan> & activeScheduleHandle) = 0;
```

Comment:
This should probably just live in the cluster impl, so we don't forget to mark it dirty when it changes.